java/json/src/main/java/com/microsoft/azure/schemaregistry/kafka/json/KafkaJsonDeserializer.java

```diff
@@ -0,0 +1,118 @@
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+
+/**
+ * Deserializer implementation for Kafka consumer, implementing Kafka Deserializer interface.
+ *
+ * @see KafkaJsonSerializer See serializer class for upstream serializer implementation
+ */
+public class KafkaJsonDeserializer<T> implements Deserializer<T> {
+    private SchemaRegistryClient client;
+    private KafkaJsonDeserializerConfig config;
+
+    /**
+     * Empty constructor used by Kafka consumer
+     */
+    public KafkaJsonDeserializer() {
+        super();
+    }
+
+    /**
+     * Configures deserializer instance.
+     *
+     * @param props Map of properties used to configure instance
+     * @param isKey Indicates if deserializing record key or value.  Required by Kafka deserializer interface,
+     *              no specific functionality has been implemented for key use.
+     *
+     * @see KafkaJsonDeserializerConfig Deserializer will use configs found in here and inherited classes.
+     */
+    public void configure(Map<String, ?> props, boolean isKey) {
+        this.config = new KafkaJsonDeserializerConfig((Map<String, Object>) props);
+
+        this.client = new SchemaRegistryClientBuilder()
+        .fullyQualifiedNamespace(this.config.getSchemaRegistryUrl())
+        .credential(this.config.getCredential())
+        .buildClient();
+    }
+
+    /**
+     * Deserializes byte array into Java object
+     * @param topic topic associated with the record bytes
+     * @param data serialized bytes, may be null
+     * @return deserialize object, may be null
+     */
+    @Override
+    public T deserialize(String topic, byte[] data) {
+        return null;
+    }
+
+    /**
+     * Deserializes byte array into Java object
+     * @param topic topic associated with the record bytes
+     * @param headers record headers, may be null
+     * @param data serialized bytes, may be null
+     * @throws JsonSerializerException Wrapped exception catchable by core Kafka producer code
+     * @return deserialize object, may be null
+     */
+    @Override
+    public T deserialize(String topic, Headers headers, byte[] data) {
+        T dataObject;
+        String schemaId;
+
+        ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        mapper.setVisibility(mapper.getVisibilityChecker().withFieldVisibility(JsonAutoDetect.Visibility.ANY));
+        try {
+            dataObject = (T) mapper.readValue(data, this.config.getJsonSpecificType());
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
General comment on printStackTrace() - Being a library package we won't know what user setup in regard to error tracking, so we should avoid this stack print. see [https://stackoverflow.com/questions/7469316/why-is-exception-printstacktrace-considered-bad-practice](url)